File: paul-beatrice-student-code-main/module-3/week-8/lecture-final/server-java/src/main/java/com/techelevator/service/ReviewService.java
package com.techelevator.service;
import com.techelevator.model.Review;
import java.util.List;
public interface ReviewService {
    Review createReview(Review review);
    Review getReviewById(int id);
    List<Review> getReviewsByProductId(int productId);
    List<Review> getAllReviews();
    Review updateReview(Review review);
    void deleteReview(int id);

}

File: paul-beatrice-student-code-main/module-3/week-8/lecture-final/client-vue/src/services/ReviewService.js
import axios from 'axios';

const reviewService = {
    getAllReviews() {
      return axios.get('/reviews');
    },
  
    getReviewById(id) {
      return axios.get(`/reviews/${id}`);
    },
  
    createReview(review) {      
      return axios.post('/reviews', review);
    },
  
    updateReview(id, review) {
      return axios.put(`/reviews/${id}`, review);
    },
  
    deleteReview(id) {
      return axios.delete(`/reviews/${id}`);
    },
  
    getReviewsByProductId(productId) {
      return axios.get(`/reviews/product/${productId}`);
    },
  };
  
  export default reviewService;